File: python/mirascope/llm/providers/google/model_id.py
"""Google registered LLM models."""

from typing import TypeAlias, get_args

from .model_info import GoogleKnownModels

GoogleModelId: TypeAlias = GoogleKnownModels | str
"""The Google model ids registered with Mirascope."""

GOOGLE_KNOWN_MODELS: set[str] = set(get_args(GoogleKnownModels))


def model_name(model_id: GoogleModelId) -> str:
    """Extract the google model name from a full model ID.

    Args:
        model_id: Full model ID (e.g. "google/gemini-2.5-flash")

    Returns:
        Provider-specific model ID (e.g. "gemini-2.5-flash")
    """
    return model_id.removeprefix("google/")

File: typescript/src/llm/providers/google/model-id.ts
/**
 * Google registered LLM models.
 */

import type { GoogleKnownModels } from '@/llm/providers/google/model-info';

/**
 * The Google model IDs registered with Mirascope.
 */
export type GoogleModelId = GoogleKnownModels | (string & {});

/**
 * Extract the Google model name from the ModelId.
 *
 * @param modelId - Full model ID (e.g. "google/gemini-2.5-flash")
 * @returns Provider-specific model ID (e.g. "gemini-2.5-flash")
 */
export function modelName(modelId: GoogleModelId): string {
  return modelId.replace(/^google\//, '');
}
